muxers/mplex/src/lib.rs

```diff
@@ -79,117 +83,142 @@ where
 /// This implementation isn't capable of detecting when the underlying socket changes its address,
 /// and no [`StreamMuxerEvent::AddressChange`] event is ever emitted.
 pub struct Multiplex<C> {
-    io: Mutex<io::Multiplexed<C>>,
+    io: Arc<Mutex<io::Multiplexed<C>>>,
```

Comment:
This is a `parkling_lot::Mutex`. I'll try if a native async one makes a difference.